src/aiortc/contrib/media.py

```diff
@@ -100,64 +103,102 @@ def player_worker(
     start_time = time.time()
 
     while not quit_event.is_set():
-        try:
-            frame = next(container.decode(*streams))
-        except (av.AVError, StopIteration) as exc:
-            if isinstance(exc, av.FFmpegError) and exc.errno == errno.EAGAIN:
-                time.sleep(0.01)
-                continue
-            if audio_track:
-                asyncio.run_coroutine_threadsafe(audio_track._queue.put(None), loop)
-            if video_track:
-                asyncio.run_coroutine_threadsafe(video_track._queue.put(None), loop)
-            break
-
-        # read up to 1 second ahead
-        if throttle_playback:
-            elapsed_time = time.time() - start_time
-            if frame_time and frame_time > elapsed_time + 1:
-                time.sleep(0.1)
-
-        if isinstance(frame, AudioFrame) and audio_track:
-            if (
-                frame.format.name != audio_format_name
-                or frame.layout.name != audio_layout_name
-                or frame.sample_rate != audio_sample_rate
-            ):
-                frame.pts = None
-                frame = audio_resampler.resample(frame)
-
-            # fix timestamps
-            frame.pts = audio_samples
-            frame.time_base = fractions.Fraction(1, audio_sample_rate)
-            audio_samples += frame.samples
-
-            audio_fifo.write(frame)
-            while True:
-                frame = audio_fifo.read(audio_samples_per_frame)
-                if frame:
-                    frame_time = frame.time
-                    asyncio.run_coroutine_threadsafe(
-                        audio_track._queue.put(frame), loop
+        if transcode:
```

Comment:
`frame` object is result of `decode()` and `packet` is result of `demux`. Because of this behavior, I need to use this condition. I don't have idea to improve this snippet.

Reply:
I prefer to keep this code, because is safer and easy to understand (each type has a different way in condition).